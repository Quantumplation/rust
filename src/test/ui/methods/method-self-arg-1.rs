// Test method calls with self as an argument cannot subvert type checking.

struct Foo;

impl Foo {
    fn bar(&self) {}
}

fn main() {
    let x = Foo;
    Foo::bar(x); //~  ERROR arguments to this function are incorrect
                 //~| expected `&Foo`, found struct `Foo`
    Foo::bar(&42); //~  ERROR arguments to this function are incorrect
                      //~| expected struct `Foo`, found integer
                      //~| expected reference `&Foo`
                      //~| found reference `&{integer}`
}
