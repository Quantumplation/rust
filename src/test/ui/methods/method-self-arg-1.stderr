error[E0308]: arguments to this function are incorrect
  --> $DIR/method-self-arg-1.rs:11:5
   |
LL |     Foo::bar(x);
   |     ^^^^^^^^^-^
   |              |
   |              expected `&Foo`, found `Foo`
   |
note: associated function defined here
  --> $DIR/method-self-arg-1.rs:6:8
   |
LL |     fn bar(&self) {}
   |        ^^^ -----
help: provide an argument of the correct type
   |
LL |     bar({&Foo});
   |     ^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/method-self-arg-1.rs:13:5
   |
LL |     Foo::bar(&42);
   |     ^^^^^^^^^---^
   |              |
   |              expected `&Foo`, found `&{integer}`
   |
note: associated function defined here
  --> $DIR/method-self-arg-1.rs:6:8
   |
LL |     fn bar(&self) {}
   |        ^^^ -----
help: provide an argument of the correct type
   |
LL |     bar({&Foo});
   |     ^^^^^^^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
